crypto/src/transaction/builder.rs

```diff
@@ -126,7 +126,24 @@ impl Builder {
     }
 
     /// Set the transaction fee in PEN.
-    pub fn set_fee(mut self, fee: u64) -> Self {
+    pub fn set_fee<R: RngCore + CryptoRng>(mut self, rng: &mut R, fee: u64) -> Self {
```

Comment:
this could be rng: mut R instead, and then people can pass an owned or boorowed RNG (since &mut R impls RngCore if R does).